imexam/ds9_viewer.py

```diff
@@ -333,44 +342,30 @@ def _set_frameinfo(self):
                                    'extname': extname,
                                    'naxis': naxis,
                                    'numaxis': numaxis,
-                                   'iscube':  iscube,
+                                   'iscube': iscube,
                                    'user_array': user_array,
                                    'mef': mef_file}
 
-    def _check_filetype(self, filename=None):
-        """check the file to see if the file is a multi-extension fits file"""
-        if not filename:
-            raise ValueError("No filename provided")
         else:
-            try:
-                mef_file = fits.getval(filename, ext=0, keyword='EXTEND')
-            except KeyError:
-                mef_file = False
-
-            # check to see if the fits file lies
-            if mef_file:
-                try:
-                    nextend = fits.getval(filename, ext=0, keyword='NEXTEND')
-                except KeyError:
-                    mef_file = False
-
-            return mef_file
+            warnings.warn("No frame loaded in viewer")
 
     def valid_data_in_viewer(self):
         """return bool if valid file or array is loaded into the viewer"""
         frame = self.frame()
 
-        if self._viewer[frame]['filename']:
-            return True
-        else:
-            try:
-                if self._viewer[frame]['user_array'].any():
-                    valid = True
-            except (AttributeError, ValueError):
-                valid = False
-                print("error in array")
+        if frame:
+            self._set_frameinfo()
 
-            return valid
+            if self._viewer[frame]['filename']:
+                return True
+
+            else:
+                try:
+                    if self._viewer[frame]['user_array'].any():
+                        return True
+                except AttributeError as ValueError:
```

Comment:
Not sure what's going on here...  Was this supposed to be `except (AttributeError, ValueError):` ?